pandas/core/series.py

```diff
@@ -4661,7 +4661,8 @@ def to_timestamp(self, freq=None, how="start", copy=True) -> "Series":
         if copy:
             new_values = new_values.copy()
 
-        assert isinstance(self.index, (ABCDatetimeIndex, ABCPeriodIndex))
+        if not isinstance(self.index, (ABCDatetimeIndex, ABCPeriodIndex)):
+            raise TypeError(f"unsupported Type {self.index}")
```

Comment:
I _think_ should also raise the TypeError for DatetimeIndex

```
>>> import pandas as pd
>>>
>>> pd.__version__
'1.1.0.dev0+1502.g3ed7dff48'
>>>
>>> rng = pd.date_range("1/1/2012", periods=5, freq="M")
>>>
>>> ts = pd.Series(np.random.randn(len(rng)), index=rng)
>>>
>>> ts
2012-01-31    0.384143
2012-02-29   -0.060017
2012-03-31   -1.506513
2012-04-30    0.898198
2012-05-31    1.035634
Freq: M, dtype: float64
>>>
>>> ts.index
DatetimeIndex(['2012-01-31', '2012-02-29', '2012-03-31', '2012-04-30',
               '2012-05-31'],
              dtype='datetime64[ns]', freq='M')
>>>
>>> ts.to_timestamp()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\simon\pandas\pandas\core\series.py", line 4665, in to_timestamp
    new_index = self.index.to_timestamp(freq=freq, how=how)
AttributeError: 'DatetimeIndex' object has no attribute 'to_timestamp'
```

Reply:
Thanks @simonjayhawkins , 
I tried figuring out where I could make this change , but what I have understood is 

error occurs at 
```python
new_index = self.index.to_timestamp(freq=freq, how=how)
```
That means it is passing this step 
```python
isinstance(self.index, (ABCDatetimeIndex, ABCPeriodIndex))
```
which implies DatetimeIndex is sub class / type of either ABCDatetimeIndex, ABCPeriodIndex , have not gone in much details. I am trying to explore more.

Can you guide me where  this change would be required if possible.